qcodes/tests/test_autoloadable_channels.py

```diff
@@ -0,0 +1,282 @@
+            name: str,
+            channel: int,
+            greeting: str,
+            existence: bool = False,
+            channel_list: 'AutoLoadableChannelList' = None,
+    ) -> None:
+
+        super().__init__(parent, name, existence, channel_list)
+        self._channel = channel
+        self._greeting = greeting
+
+        self.add_parameter(
+            "hello",
+            get_cmd=f":INST:CHN{self._channel}:HLO"
+        )
+
+    def _create(self)->None:
+        """Create the channel on the instrument"""
+        self.parent.root_instrument.write(
+            f":INST:CHN:ADD {self._channel}, {self._greeting}")
+
+    def _remove(self)->None:
+        """Remove the channel from the instrument"""
+        self.write(f":INST:CHN:DEL {self._channel}")
+
+
+class DummyInstrument(Instrument):
+    """
+    This dummy instrument allows the creation and deletion of
+    channels
+    """
+
+    def __init__(self, name: str)->None:
+        super().__init__(name)
+
+        self._backend = MockBackend()
+
+        self.add_parameter(
+            "channel_catalog",
+            get_cmd=":INST:CHN:CAT",
+        )
+
+        channels = AutoLoadableChannelList(
+            self, "channels", SimpleTestChannel, channels_to_skip=["5"])
+        self.add_submodule("channels", channels)
+
+    def write_raw(self, cmd: str)->None:
+        self._backend.send(cmd)
+
+    def ask_raw(self, cmd: str)->str:
+        return self._backend.send(cmd)
+
+
+@pytest.fixture(scope='function')
+def dummy_instrument():
+    instrument = DummyInstrument("instrument")
+    yield instrument
+    instrument.close()
+
+
+def test_sanity(dummy_instrument):
+    """
+    Test the basic functionality of the auto-loadable channels, without using
+    the auto-loadable channels list
+    """
+    # Test that we are able to discover instruments automatically
+    channels = SimpleTestChannel.load_from_instrument(dummy_instrument)
+    assert len(channels) == 4
+    assert channels[0].hello() == "Hello from channel 1"
+    assert channels[1].hello() == "Hello from channel 2"
+    assert channels[2].hello() == "Hello from channel 4"
+    assert channels[3].hello() == "Hello from channel 5"
+    # Test that we can generate a new instance of the channels without
+    # conflicting names
+    new_channel_kwargs = SimpleTestChannel._get_new_instance_kwargs(
+        dummy_instrument)
+
+    new_channel = SimpleTestChannel(dummy_instrument, **new_channel_kwargs)
+    # Instrument IO through the newly instantiated channel should raise an
+    # exception before actually creating the channel on the instrument
+    with pytest.raises(
+            RuntimeError,
+            message="Object does not exist (anymore) on the instrument"):
+
+        new_channel.hello()
+
+    # After creating the channel we should be able to talk to the instrument.
+    new_channel.create()
+    assert new_channel.hello() == "Hello from channel 3"
```

Comment:
spaces for readability:
```suggestion
    assert new_channel.hello() == "Hello from channel 3"
    
```

Reply:
Is it me, or do I not see a difference between the lines.

Reply:
extra newline after the assert statement :)